### ISSUE TITLE
Equações não estão renderizando corretamente no iPhone

### ISSUE THREAD

**Issue by carolinatanigushi** [scieloorg/opac#1381]
### Descrição do problema
(relacionado ao ticket #1361)

Equações não estão renderizando corretamente quando o acesso é feito pelo Safari ou Chrome de um iPhone. Esse problema não acontece no Opera nem em celulares rodando Android.

OBS: "Esse problema não foi introduzido pelo responsivo mas nós só pudermos ver quando o responsivo começou a funcionar".

### Passos para reproduzir o problema
1. Acesse um artigo com equações por um iPhone. Por exemplo: [Conceitos básicos de radiômetros de substituição elétrica para medidas da Irradiância Solar Total](https://homolog.opac.scielo.br/article/rbef/2019.v41n2/e20180220/pt/)
2. Role a página até a equação (1), na seção 3.

### Comportamento esperado
Renderizar as equações corretamente.

### Screenshots ou vídeos
**Safari:**
![safari](https://user-images.githubusercontent.com/12699183/61960285-91a7dd80-af9b-11e9-834f-86a00a4f2a03.png)

**Chrome:**
![chrome](https://user-images.githubusercontent.com/12699183/61960308-979dbe80-af9b-11e9-8f6c-71cc64d87678.png)

### Ambiente utilizado
Aparelho celular iPhone 8, iOS 12.3.2, testado nos navegadores:
- Safari;
- Chrome; e
- Opera.

**Comment by joffilyfe** [comment]
Oi @carolinatanigushi eu fiz uma análise inicial sobre o caso reportado por você e observei que o problema ocorre apenas no servidor de homologação.

Acessando a URL referenciada pelo ticket a partir o Safari eu tive essas duas visões:

**Modo desktop**
![Screen Shot 2019-09-23 at 17 27 36](https://user-images.githubusercontent.com/4604104/65460106-9536d400-de27-11e9-840d-57310ac6ab76.png)

**Modo mobile**
![Screen Shot 2019-09-23 at 17 28 06](https://user-images.githubusercontent.com/4604104/65460108-9536d400-de27-11e9-8790-413352ac0eb3.png)

O mesmo comportamento não ocorre quando acesso o artigo no servidor de produção:
![Screen Shot 2019-09-23 at 17 30 26](https://user-images.githubusercontent.com/4604104/65460239-daf39c80-de27-11e9-8c85-108f4e1e7387.png)

Existem sim outros problemas como:
1) Label da fórmula sobrepondo a renderização
2) Fórmulas muito grandes ficam ilegíveis em resoluções pequenas;

Na sua opinião qual seria o melhor desfecho para este ticket?

**Comment by carolinatanigushi** [comment]
Olá, @joffilyfe

Conversando com o @alexxxmendonca, entende-se que, por mais que só esteja ocorrendo somente no homolog, esse problema deve ser resolvido, pois a homologação precisa ser o espelho da produção. Não é bom que os dois sites fiquem diferentes, pois isso pode causar impacto, por exemplo, no trabalho de controle de qualidade da unidade de produção.

**Comment by joffilyfe** [comment]
Após realizar vários testes pude reproduzir o problema no ambiente de homologação. O problema se mostra intermitente e logo após o segundo carregamento da página a fórmula é renderizada corretamente (a depender do cache).

### Reprodução do problema
Consegui estabelecer os passos necessários para reproduzir o problema no ambiente de desenvolvimento, segue:
1. Abrir o Safari;
2. Limpar o cache do navegador;
3. Abrir uma janela anônima;
4. Acessar o link `/article/rbef/2019.v41n2/e20180220/pt` (+host);
5. Visualizar a fórmula quebrada.
6. Exemplo em vídeo https://www.youtube.com/watch?v=dMysePmiMe8&feature=youtu.be

Também consegui estabelecer um método para que a página seja renderizada corretamente, segue:
1. Desativar o script `badge.dimensions.ai/badge.js` na source do projeto;
2. Abrir o Safari;
3. Limpar o cache do navegador;
4. Abrir uma janela anônima;
5. Acessar o link `/article/rbef/2019.v41n2/e20180220/pt` (+host);
6. Visualizar a fórmula sendo exibida corretamente.
7. Exemplo em vídeo https://www.youtube.com/watch?v=AATuaOyQwbQ&feature=youtu.be

O comportamento acima é curioso porque sugere que o javascript do `dimensions` tem alguma interferência no processo de renderização do MathJax, não posso afirmar que seja isso, o que pude verificar é que uma `web font/estilo` não é carregada em tempo hábil e portanto o javascript do mathjax não consegue calcular e definir as dimensões da fórmula.

### Resolução parcial

Utilizando a versão `3.0.0` as renderizações passaram a funcionar em todos os browsers e resoluções de celular. Esta versão parte de uma total reescrita da biblioteca mas algumas funcionalidades foram removidas [(ex: renderização CSS-HTML)](http://docs.mathjax.org/en/latest/upgrading/v2.html#changes-in-input-and-output-jax) e a biblioteca está em `WIP`.

**Comment by joffilyfe** [comment]
@carolinatanigushi a correção está disponível no ambiente de homologação, você poderia conferir por favor?

**Comment by carolinatanigushi** [comment]
> Oi @carolinatanigushi eu fiz uma análise inicial sobre o caso reportado por você e observei que o problema ocorre apenas no servidor de homologação.
> 
> Acessando a URL referenciada pelo ticket a partir o Safari eu tive essas duas visões:
> 
> **Modo desktop**
> ![Screen Shot 2019-09-23 at 17 27 36](https://user-images.githubusercontent.com/4604104/65460106-9536d400-de27-11e9-840d-57310ac6ab76.png)
> 
> **Modo mobile**
> ![Screen Shot 2019-09-23 at 17 28 06](https://user-images.githubusercontent.com/4604104/65460108-9536d400-de27-11e9-8790-413352ac0eb3.png)
> 
> O mesmo comportamento não ocorre quando acesso o artigo no servidor de produção:
> ![Screen Shot 2019-09-23 at 17 30 26](https://user-images.githubusercontent.com/4604104/65460239-daf39c80-de27-11e9-8c85-108f4e1e7387.png)
> 
> Existem sim outros problemas como:
> 
> 1. Label da fórmula sobrepondo a renderização
> 2. Fórmulas muito grandes ficam ilegíveis em resoluções pequenas;
> 
> Na sua opinião qual seria o melhor desfecho para este ticket?

**Comment by gustavofonseca** [comment]
@joffilyfe, o problema se manifesta para qualquer formato de renderização ou apenas SVG?

**Comment by joffilyfe** [comment]
> @joffilyfe, o problema se manifesta para qualquer formato de renderização ou apenas SVG?

Eu só consegui reproduzir utilizando o SVG. Um dado curioso é que nas issues do mathjax algumas pessoas reclamaram do mesmo problema (https://github.com/mathjax/MathJax/issues/955) mas eu não consegui achar a solução.

**Comment by carolinatanigushi** [comment]
Segue um gif do erro:

![FilefromiOS](https://user-images.githubusercontent.com/12699183/68236551-216e1680-ffe4-11e9-987d-ece72257912b.gif)

Tem também um vídeo um pouco mais longo, em que testei vários artigos (em alguns casos as equações carregavam normalmente, em outros não): <[https://youtu.be/OE55PXf5jOA](https://youtu.be/OE55PXf5jOA)>

**Comment by robertatakenaka** [comment]
@carolinatanigushi será que você conseguiria fazer um teste? Há uns artigos e não me engano em eLife ou PLoS que tenham artigos com fórmulas. Será que você consegue encontrar exemplos de fórmulas que funcionem ou não nestes sites? Assim talvez a gente se inspire em encontrar a solução. Será que eles não teriam o mesmo problema? ou a mesma situação?

**Comment by carolinatanigushi** [comment]
@robertatakenaka 

Dei uma olhada na eLife e no [arXiv ](https://arxiv.org/help/mathjax) (na PLOS as equações são capturadas como imagem).

Não consegui muita coisa [olhando o arXiv](https://youtu.be/Tb6Dg9zAFv4) até porque há apenas o resumo, mas nenhum dos preprints que eu testei deu erro.

No que diz respeito à eLife, não sei se é apenas uma impressão minha, nem se ajuda em alguma coisa, mas percebi que parece levar um tempo maior para carregar as equações; além disso, o site carrega os artigos seção por seção. Não consegui replicar o erro em nenhum dos artigos que testei.

Aqui tem um vídeo em que vejo todo o artigo "[Length regulation of multiple flagella that self-assemble from a shared pool of components](https://elifesciences.org/articles/42599)": [https://youtu.be/V-xGHps-4FI](https://youtu.be/V-xGHps-4FI)

**Comment by alexxxmendonca** [comment]
O problema foi resolvido:

![Image from iOS](https://user-images.githubusercontent.com/4059679/121411400-f8d59d80-c939-11eb-972b-7c36b5310e23.png)

O truncamento de informações que observa-se aqui se deve ao tamanho da tela do dispositivo. Isso corrige-se ao mudar a orientação da tela para a horizontal:

![Image from iOS (1)](https://user-images.githubusercontent.com/4059679/121411531-1c98e380-c93a-11eb-84f6-82c8d5304cf4.png)

De qualquer forma, se formos resolver esta nova questão, melhor ser tratado em um outro ticket.

Como a motivação original deste ticket foi resolvida, estou fechando-o.
